featuretools/entityset/entityset.py

```diff
@@ -803,6 +803,7 @@ def normalize_dataframe(self, base_dataframe_name, new_dataframe_name, index,
 
         new_dataframe2 = new_dataframe. \
             drop_duplicates(index, keep='first')[selected_columns]
+        new_dataframe2 = new_dataframe2.dropna(subset=[index])
```

Comment:
doing the drop here might cause a scenario where `make_secondary_time_index` logic reintroduces null index values but I'm not positive